libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -691,7 +699,29 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		for (size_t i = 0; i < arguments.size(); ++i)
 			args += convert(*arguments[i], *parameterTypes[i]).stackSlots();
 
-		if (functionDef)
+		if (_functionCall.annotation().kind == FunctionCallKind::StructConstructorCall)
```

Comment:
Actually I think this should not be inside the `	case FunctionType::Kind::Internal:`-case - not even inside the switch. Please take a look how it is done in the old code generator. Some lines above we do virtual function lookup and check for 'bound' - this is all irrelevant here.

If you pull it out, and just run a loop over `arguments`, the problem with the multi-slot types will also resolve itself easily, I think.